feat: add function to store data

jerrylin03 opened the pull request:
Add function to prompt for user data, also change all storage from CSV to JSON file format

jerrylin03 committed:
feat: add function to store data

Add function to prompt for user data, also change all storage from CSV to JSON file format

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,94 @@ def fetch_properties(state_code):
-    else:
+    elif args.function == "schools":
         fetch_top_school_districts(args.state_code)
+    elif args.function == "forms":
```
what does forms mean?

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,94 @@ def fetch_properties(state_code):
+        'gross_income': float(input("Enter your gross income $: ")),
+        'current_debt': float(input("Enter your current debt $: ")),
+        'property_management_fees': float(input("Enter property management fees %: ")),
+        'maintenance_costs': float(input("Enter maintenance costs: ")),
```
make this a percent of rental income

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,94 @@ def fetch_properties(state_code):
+        'current_debt': float(input("Enter your current debt $: ")),
+        'property_management_fees': float(input("Enter property management fees %: ")),
+        'maintenance_costs': float(input("Enter maintenance costs: ")),
+        'property_taxes': float(input("Enter property taxes: ")),
```
todo: compute this based on state - so no need to enter manual State and then its property tax

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,94 @@ def fetch_properties(state_code):
+        # Create a new user_data dictionary if the file doesn't exist
+        user_data = {}
+
+    # Check if the username is already in the database
+    if username in user_data:
+        print(f"Hello, {username}!")
+    else:
+        print(f"New user {username}")
```
Put this logic in the try catch - its already doing this check essentially

jerrylin03 committed:
fix: revise from PR comments

- refactor forms variable to profile
- change profile input to be percent of income
- remove property tax input
- move profile check inside try catch

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,93 @@ def fetch_properties(state_code):
+
+    print(f"User data for {username} has been updated.")
+
 if __name__ == "__main__":
```
nit: suggest using [Click](https://click.palletsprojects.com/en/8.1.x/) for cmd line interface

jerrylin03 replied:
Using the module argparse to handle CLI commands, but I'm printing out info for more verbose output

jlin332 commented on scanner_cli.py:
```diff
@@ -23,37 +26,40 @@ def fetch_top_school_districts(state_code):
+    """ Function to get the newest 42 properties for sale by state saved to a json file """
     api_url = "https://us-real-estate.p.rapidapi.com/v3/for-sale"
 
     querystring = {"state_code":"VA","sort":"newest","offset":"0"}
```
nit: querystring should use state_code as input

jlin332 commented on scanner_cli.py:
```diff
@@ -5,15 +5,18 @@
 import csv
 from pip._vendor import requests
 
+API_KEY = ""
```
API_KEY can be an enviornment variable - `os.getenv("API_KEY")` and just do `export API_KEY=xxxx` in your terminal shell

jerrylin03 replied:
Yeah, I can change this. Was like this because initially you said to just have the key hardcoded in for now

jlin332 commented on scanner_cli.py:
```diff
@@ -63,40 +69,93 @@ def fetch_properties(state_code):
+        print(f"New user {username}")
+
+    # Ask for and save user input
+    user_data[username] = {
```
if username already exists do we want to enter in all of this information again?

jerrylin03 replied:
The reason I made it so that we would re-enter this info again was so that we have the option to update the fields. I think what I will do is output the profile if it exists and ask if they wanted to update the fields.

jerrylin03 committed:
fix: revise PR comments

- change query string to use input
- load API key from .env
- prompt for user profile update

jerrylin03 committed:
refactor: pull from .env

change API variables such as URL, host, key to pull from .env
